### PR TITLE
xds: support ring_hash as the endpoint-level LB policy

### PR DESCRIPTION
**Pull request by voidzcy**
To be merged after #7943.

There two main parts of this change:
- Changed the CDS LB policy to accept "ring_hash" as the endpoint LB policy from CDS updates. This configuration is directly passed to its child policy (aka, ClusterResolverLoadBalancer) in its config.
  - Note, for underlying clusters of an aggregate cluster, its endpoint-level LB policy is configured by the top-level cluster configuration.
- Changed ClusterResolverLoadBalancer to generate different LB configs for its downstream LB policies. Depending on the endpoint-level LB policies, the downstream LB tree hierarchy will be different.
  - If the endpoint-level LB policy is "round_robin", the downstream LB policy hierarchy is: PriorityLB -> ClusterImplLB -> WeightedTargetLB -> RoundRobinLB (same as existing).
  - If the endpoin-level LB policy is "ring_hash", the downstream LB policy hierarchy is: PriorityLB -> ClusterImplLB -> RingHashLB. That is, endpoints in different localities are mixed together for load balancing. Each endpoints will be attached with a weight value that is the product of its endpoint-level weight and the weight of the locality it belongs to.

### CODE REVIEW

**Comment by voidzcy** [comment]
This can be reviewed and merged now.

File: xds/src/main/java/io/grpc/xds/CdsLoadBalancer2.java [hunk 1]
```diff
@@ -181,15 +182,13 @@ private void handleClusterDiscovered() {
...
         return;
       }
-      String endpointPickingPolicy = root.result.lbPolicy();
-      LoadBalancerProvider localityPickingLbProvider =
-          lbRegistry.getProvider(XdsLbPolicies.WEIGHTED_TARGET_POLICY_NAME);  // hardcoded
-      LoadBalancerProvider endpointPickingLbProvider =
-          lbRegistry.getProvider(endpointPickingPolicy);
+      LoadBalancerProvider lbProvider = lbRegistry.getProvider(root.result.lbPolicy());
+      Object lbConfig = null;
+      if (root.result.lbPolicy().equals("ring_hash")) {
```
Comment (dapengzhang0) [inline xds/src/main/java/io/grpc/xds/CdsLoadBalancer2.java:187]: What about make `CdsUpdate.lbPolicy` enum type {RING_HASH, ROUND_ROBIN}?

**Review by dapengzhang0 — COMMENTED** [review]
LGTM

File: xds/src/main/java/io/grpc/xds/CdsLoadBalancer2.java (hunk 1 above)
Comment (voidzcy) [reply xds/src/main/java/io/grpc/xds/CdsLoadBalancer2.java]: Sure.

**Review by dapengzhang0 — APPROVED** [review]

**Pull request merged**